src/StagedProposalProcessorSetup.sol

```diff
@@ -37,55 +46,68 @@ contract StagedProposalProcessorSetup is PluginUpgradeableSetup {
+        // a delegate call which will still make `msg.sender` to be sub-plugin on SPP,
+        // so as default, we set trusted forwarder = address(0), but grantee of
+        // `SET_TRUSTED_FORWARDER_PERMISSION` can anytime set the actual address.
+        // Setting a user's passed trusted forwarder below is dangerous in case plugin
+        // installer is malicious.
         plugin = IMPLEMENTATION.deployUUPSProxy(
-            abi.encodeCall(
-                SPP.initialize,
-                (IDAO(_dao), address(trustedForwarder), stages, metadata, targetConfig)
-            )
+            abi.encodeCall(SPP.initialize, (IDAO(_dao), address(0), stages, pluginMetadata, targetConfig))
         );
 
         PermissionLib.MultiTargetPermission[]
-            memory permissions = new PermissionLib.MultiTargetPermission[](3);
+            memory permissions = new PermissionLib.MultiTargetPermission[](5);
 
         permissions[0] = PermissionLib.MultiTargetPermission({
-            operation: PermissionLib.Operation.GrantWithCondition,
+            operation: PermissionLib.Operation.Grant,
             where: plugin,
-            // TODO:GIORGI If ANY_ADDR permission, PM not allowing it unless condition is set, do we want to leave this as it is in PM ?
-            who: ANY_ADDR,
-            condition: address(new AlwaysTrueCondition()),
-            permissionId: ADVANCE_PROPOSAL_PERMISSION_ID
+            who: _dao,
+            condition: PermissionLib.NO_CONDITION,
+            permissionId: UPDATE_STAGES_PERMISSION_ID
         });
 
         permissions[1] = PermissionLib.MultiTargetPermission({
+            operation: PermissionLib.Operation.Grant,
+            where: _dao,
+            who: plugin,
+            condition: PermissionLib.NO_CONDITION,
+            permissionId: EXECUTE_PERMISSION_ID
+        });
+
+        permissions[2] = PermissionLib.MultiTargetPermission({
             operation: PermissionLib.Operation.Grant,
             where: plugin,
             who: _dao,
             condition: PermissionLib.NO_CONDITION,
-            permissionId: UPDATE_STAGES_PERMISSION_ID
+            permissionId: SET_TRUSTED_FORWARDER_PERMISSION_ID
         });
 
-        permissions[2] = PermissionLib.MultiTargetPermission({
+        permissions[3] = PermissionLib.MultiTargetPermission({
             operation: PermissionLib.Operation.Grant,
-            where: _dao,
-            who: plugin,
+            where: plugin,
+            who: _dao,
             condition: PermissionLib.NO_CONDITION,
-            permissionId: DAO(payable(_dao)).EXECUTE_PERMISSION_ID()
+            permissionId: SET_TARGET_CONFIG_PERMISSION_ID
+        });
+
+        permissions[4] = PermissionLib.MultiTargetPermission({
+            operation: PermissionLib.Operation.Grant,
+            where: plugin,
+            who: _dao,
+            condition: PermissionLib.NO_CONDITION,
+            permissionId: UPDATE_METADATA_PERMISSION_ID
```

Comment:
Does the SPP have metadata that needs to be used anywhere? 
It's fine, just thinking that the metadata was meant to be on the subplugins

Reply:
Yes, SPP has as well. no worries.